samples/alarmbot-es6-botframework-webchat/src/webChatAdapter.js

```diff
@@ -1,25 +1,46 @@
-const activitiesById = {};
+import { ConnectionStatus } from 'botframework-webchat';
+import { Subject, Observable, BehaviorSubject } from 'rxjs';
 
 export class WebChatAdapter {
+    constructor() {
+        this.activity$ = new Subject();
+        this.botConnection = {
+            connectionStatus$: new BehaviorSubject(ConnectionStatus.Online),
+            activity$: this.activity$.share(),
+            end() {
+                debugger
+            },
+            postActivity: activity => {
+                const id = Date.now().toString();
+                return Observable.fromPromise(this
+                    .onReceive(Object.assign({}, activity, {
+                        id,
+                        conversation: { id: "bot" },
+                        channelId: 'WebChat'
+                    }))
+                    .then(() => id)
+                )
+            }
+        }
+    }
+
     post(activities) {
-        activitiesById[activities[0].replyToId] = activities;
-        activities.forEach(activity => Object.assign(activity, {id: activity.replyToId, from: {id: "The Bot"}}));
-        return Promise.resolve(activities);
+        let sentActivities = activities.map(activity => Object.assign({}, activity, {
+            id: Date.now().toString(),
+            channelId: 'WebChat',
+            conversation: { id: 'bot' },
+            from: { id: 'bot' }
+        }));
+
+        sentActivities.forEach(activity => this.activity$.next(activity));
```

Comment:
I should have caught this in the previous code review but you're not handling the 'delay' activity we made up.  This isn't an official activity type YET but Dan is considering adding it given that it's natively supported in channels like Kik and Line. We currently model 'delay' in the adapter by just doing:  setTimeout(() => { next-activity }, activity.value)

Reply:
My preference would be to flow this into Web Chat and letting it handle it natively.

Reply:
Sounds good...